Improve boolean indexing docs

madisongoggin opened the pull request:
- [ X] closes #42384
- [ X] tests added / passed
- [ X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ X] whatsnew entry

madisongoggin committed:
DOC: Improve Boolean Vector Docs #42384

Add missing boolean vector (`^`). Create table for quick reference. Solves #42384.

madisongoggin committed:
DOC: Improve Boolean Vector Docs #42384

Fix table issues (incorrect format).

madisongoggin committed:
DOC: Update latest `whatsnew`

Add Doc Fix for #42384 in `whatsnew`.

mroeschke commented on doc/source/whatsnew/v1.4.0.rst:
```diff
@@ -32,7 +32,7 @@ Other enhancements
 -  Additional options added to :meth:`.Styler.bar` to control alignment and display (:issue:`26070`)
 - :meth:`Series.ewm`, :meth:`DataFrame.ewm`, now support a ``method`` argument with a ``'table'`` option that performs the windowing operation over an entire :class:`DataFrame`. See :ref:`Window Overview <window.overview>` for performance and functional benefits (:issue:`42273`)
--
+- The docs for Boolean vector indexing were enhanced (:issue:`42384`)
```
Don't need a whatsnew entry for doc improvements

mroeschke commented on doc/source/user_guide/indexing.rst:
```diff
@@ -881,6 +894,7 @@ Using a boolean vector to index a Series works exactly as in a NumPy ndarray:
    s
    s[s > 0]
    s[(s < -1) | (s > 0.5)]
+   s[(s < -1) ^ (s > 0.5)]
```
Could you add an example for `&`

madisongoggin committed:
Revert "DOC: Improve Boolean Vector Docs #42384"

This reverts commit 19b88c34c8c3249982823b75496ff2036e63872f.

mroeschke commented on doc/source/user_guide/indexing.rst:
```diff
@@ -867,7 +867,20 @@ Boolean indexing
 
 Another common operation is the use of boolean vectors to filter the data.
-The operators are: ``|`` for ``or``, ``&`` for ``and``, and ``~`` for ``not``.
+The operators are: ``|`` for ``or``,  ``^`` for ``bitwise or``, ``&`` for ``and``, and ``~`` for ``not``. 
```
This sentence seem redundant now with the table you provided (which is a lot better IMO)

madisonestabrook replied:
I reverted back to before my whatsnew commit. Then, I added and committed an example for  `&` and  removed the redundant sentence.

Sorry about the whatsnew! I saw that in the PR template and thought - for whatever reason - it was required.

madisongoggin committed:
Remove redundant sentence

Removed the  redundant sentence, which repeated information contained in the table.

madisongoggin committed:
DOC: Add an `&` Example

Add an example for "and" (`&`)

madisongoggin committed:
DOC: flip backwards comparison operator

mzeitlin11 commented on doc/source/user_guide/indexing.rst:
```diff
@@ -866,8 +866,15 @@ Boolean indexing
-The operators are: ``|`` for ``or``, ``&`` for ``and``, and ``~`` for ``not``.
+Another common operation is the use of boolean vectors to filter the data. Below is a table containing available boolean vectors.
+
+| Boolean Vector | Description   |
```
Should this be operator instead of Boolean Vector? Same in the sentence above.

mroeschke commented:
Thanks for the PR, but it appears to have gone stale. This was close, but if you could push some small updates based on the review we would be happy to accept this PR. Closing in the meantime.
